hepcrawl/spiders/alpha_spider.py

```diff
@@ -0,0 +1,175 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2015 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Spider for ALPHA."""
+
+from __future__ import absolute_import, print_function
+
+import os
+import re
+import sys
+
+from urlparse import urljoin
+
+from scrapy import Request, Selector
+from scrapy.spiders import CrawlSpider
+
+from ..items import HEPRecord
+from ..loaders import HEPLoader
+
+
+class AlphaSpider(CrawlSpider):
+
+    """Alpha crawler
+    Scrapes theses metadata from Alpha experiment web page.
+    http://alpha.web.cern.ch/publications#thesis
+
+    1. parse() iterates through every record on the html page and yields
+       a HEPRecord.
+
+
+    Example usage:
+    scrapy crawl alpha -s "JSON_OUTPUT_DIR=tmp/"
+    scrapy crawl alpha -a source_file=file://`pwd`/tests/responses/alpha/test_10.htm -s "JSON_OUTPUT_DIR=tmp/"
+
+    TODO:
+    * Author names are not comma separated in the html page. Because of this,
+      possible multiple surnames are not correctly recognised.
+
+    Happy crawling!
+    """
+
+    name = 'alpha'
+    start_urls = ["http://alpha.web.cern.ch/publications#thesis"]
+    domain = "http://alpha.web.cern.ch/"
+    itertag = "//div[@class = 'node node-thesis']"
+
+    def __init__(self, source_file=None, *args, **kwargs):
+        """Construct Alpha spider"""
+        super(AlphaSpider, self).__init__(*args, **kwargs)
+        self.source_file = source_file
+
+    def start_requests(self):
+        """You can also run the spider on local test files"""
+        if self.source_file:
+            yield Request(self.source_file)
+        elif self.start_urls:
+            for url in self.start_urls:
+                yield Request(url)
+
+    def split_fullname(self, author):
+        """If we want to split the author name to surname and given names.
+        """
+        import re
+        fullname = author.split()
+        # CAVEAT: Here we are *assuming* surname comes last
+        surname = fullname[-1]
+        given_names = " ".join(fullname[:-1])
+        return surname, given_names
+
+    def has_numbers(self, s):
```

Comment:
This function can be added to `utils.py` since it does not use the spider instance `self` for anything and thus is not tied to it.